Add recipe for xmltokf

paddymcall opened the pull request:
### Brief summary of what the package does

`xmltokf.el` provides XML tokenizer functions that don’t have any side
effects.  This makes it easier to program some XML editing functions.

### Direct link to the package repository

https://github.com/paddymcall/xmltokf.el

### Your association with the package

I'm the maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  I read your README which goes into good detail.

> It wraps and relies on the functionality in the GNU/Emacs xmltok.el library

Have you thought about upstreaming this?  The major issue you describe, i.e.

> If your XML files are big enough, Emacs may decide to interrupt the process in buffer A to do some work in buffer B.

...seems like a serious issue, and might even warrant a bug report, if I'm following correctly.

In the meantime, here's a quick first pass of the code -

⸺ `xmltokf.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- xmltokf.el#L294: Triple semicolons `;;;` are usually for section headings
- xmltokf.el#L599: Triple semicolons `;;;` are usually for section headings
- xmltokf.el#L746: Triple semicolons `;;;` are usually for section headings
- xmltokf.el#L676: No `format` required; `warn` takes an f-string
- xmltokf.el#L685: No `format` required; `warn` takes an f-string
```

⸺ Package and license:
- xmltokf.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)
- Reminder: if you haven't, ensure GitHub release v0.1 is up-to-date with your current code and `Package-Version`

paddymcall commented:
Thanks for the feedback!  I didn't know about melpazoid, but I've added the workflow and it seems there are no more complaints (I've changed the version to v0.2): 

https://github.com/paddymcall/xmltokf.el/actions/runs/12545533167

I'll raise the issue I describe in the README upstream, I think the fix should be rather easy.  But in any case this library is not supposed to be a replacement for anything in Emacs, it just adds some simple and atomic XML tokenizer and editing functions.  I use this library in a few other elisp projects, and it would be helpful to have it in melpa.

riscy commented:
> I didn't know about melpazoid

If it was useful I'm glad - melpazoid is my own set of scripts and still requires a human in the loop, but whenever I *don't* link to it, people ask what I was using. :)

> no more complaints

Same here - code looks great.

The last issue I can see - and apologies for not catching it earlier - is that your `;;; Commentary` is malformed; if I run `(lm-commentary)` on your file I only get:
```
;;; Commentary:

;; Uses xmltok without setting global variables (functional, sort of).

;; To run the tests: make test
```
i.e. everything under `;;; Usage:` is omitted because the `;;; Usage:` token delineates a section that is separate from your Commentary. This one is important because this is what the MELPA index will use to describe your package.

paddymcall commented:
Thank you for the hint, I've corrected the problem in the latest version now.

riscy commented:
Thank you!

In case you haven't, ensure GitHub release v0.2.3 is up-to-date with your current code and `Package-Version`
